packages/repository/src/repositories/legacy-juggler-bridge.ts

```diff
@@ -184,9 +207,31 @@ export class DefaultCrudRepository<T extends Entity, ID>
     );
   }
 
+  /**
+   * @deprecated
+   * Function to create a belongs to accessor
+   *
+   * Use `this.createBelongsToAccessorFor()` instaed
+   *
+   * @param relationName Name of the relation defined on the source model
+   * @param targetRepo Target repository instance
+   */
   protected _createBelongsToAccessorFor<Target extends Entity, TargetId>(
     relationName: string,
     targetRepoGetter: Getter<EntityCrudRepository<Target, TargetId>>,
+  ): BelongsToAccessor<Target, ID> {
+    return this.createBelongsToAccessorFor(relationName, targetRepoGetter);
```

Comment:
Shouldn't it make sense to log the deprecation warning ?
What's the overall project strategy to deal with this ?

Reply:
Good question. We don't have a deprecation strategy yet.

I guess the question is how much we want to force our users to update their code.

A slow process:
 1. Soft deprecations (docs only)
 2. Hard deprecation (warnings printed at runtime) - maybe wait until the next semver-major?
 3. API removed (in a semver-major release)

A faster process:
1. Hard deprecation (warnings printed at runtime)
2. API removed (in a semver-major release)

In the past, we were using the fast process. From what I heard about React, their users love their slower process for API deprecations, maybe we can do the same in LoopBack?

Implementation wise, we have had good experience using [depd](https://www.npmjs.com/package/depd) module for dealing with deprecation warnings in LoopBack 2.x/3.x

Reply:
On the second thought, since this is a simple rename that has a trivial upgrade path, I am fine with a hard deprecation from the beginning.